src/components/events/ReportModal.jsx

```diff
@@ -1,26 +1,43 @@
 import React, { useState } from "react";
+import axios from "axios";
 import { XIcon } from "../icons/Icons";
 
-export default function ReportModal({ title, onDismiss }) {
+export default function ReportModal({ title, reportData, onDismiss }) {
   const [value, setValue] = useState(null)
   // eslint-disable-next-line no-unused-vars
-  const [message, setMessage] = useState(null)
+  const [reason, setReason] = useState("")
+
+  reportData.reason = (value === 'Other' ? reason : value);
+
+    const onClick = async () => {
+      try {
+        await axios.post(
+          `${process.env.REACT_APP_API_URL}/api/deal/report`,
+          { dealId: reportData.dealId, reason: reportData.reason },
+        );
+        console.log("successfully reported deal", reportData);
+        onDismiss();
```

Comment:
If a user doesn't choose an option and just clicks report. Nothing happens and then the dialog closes. Not sure if its a blocker but for the future if they don't choose an option an error message should pop up under urging a user to choose an option.